src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperDescriptorFactory.cs

```diff
@@ -83,13 +83,18 @@ public static class TagHelperDescriptorFactory
             bool designTime)
         {
             TagHelperUsageDescriptor typeUsageDescriptor = null;
+            string outputElementHint = null;
 
-#if !DNXCORE50
             if (designTime)
             {
+                var outputElementHintAttribute =
+                    typeInfo.GetCustomAttribute<OutputElementHintAttribute>(inherit: false);
+
+                outputElementHint = outputElementHintAttribute?.OutputElement;
```

Comment:
I'd be more tempted to have a whole property called `DesignTimeStuff` that then had properties for each of the design-time only parts of the descriptor, which for now might be just the hint.